File: PRO/practica45/practica45.java
import java.io.*;
import java.util.ArrayList;
import java.util.Scanner;

public class practica45 {
    private static ArrayList<String[]> usuarios = new ArrayList<>();

    private static void cargarUsuarios() {
        try {
            BufferedReader br = new BufferedReader(new FileReader("usuarios.txt"));
            String linea;
            while ((linea = br.readLine()) != null) {
                String[] userData = linea.split(", ");
                usuarios.add(userData);
            }
            br.close();
        } catch (IOException e) {
            System.out.println("Error al cargar los usuarios.");
        }
    }

    private static void guardarUsuarios() {
        try {
            BufferedWriter bw = new BufferedWriter(new FileWriter("usuarios.txt"));
            for (String[] usuario : usuarios) {
                bw.write(String.join(", ", usuario));
                bw.newLine();
            }
            bw.close();
        } catch (IOException e) {
            System.out.println("Error al guardar los usuarios.");
        }
    }

    private static void agregarUsuario(Scanner sc) {
        System.out.print("Ingrese el DNI del nuevo usuario: ");
        String dni = sc.nextLine();
        System.out.print("Ingrese el nombre del nuevo usuario: ");
        String nombre = sc.nextLine();
        System.out.print("Ingrese los apellidos del nuevo usuario: ");
        String apellidos = sc.nextLine();
        System.out.print("Ingrese el correo electrónico del nuevo usuario: ");
        String correo = sc.nextLine();
        String[] nuevoUsuario = {dni, nombre, apellidos, correo};
        usuarios.add(nuevoUsuario);
        guardarUsuarios();
        System.out.println("Usuario agregado correctamente.");
    }

    private static void eliminarUsuario(Scanner sc) {
        System.out.print("Ingrese el DNI del usuario que desea eliminar: ");
        String dni = sc.nextLine();
        boolean encontrado = false;
        for (String[] usuario : usuarios) {
            if (usuario[0].equals(dni)) {
                usuarios.remove(usuario);
                guardarUsuarios();
                System.out.println("Usuario eliminado correctamente.");
                encontrado = true;
                break;
            }
        }
        if (!encontrado)
            System.out.println("Usuario no encontrado.");
    }

    private static void buscarUsuario(Scanner sc) {
        System.out.print("Ingrese el DNI del usuario que desea buscar: ");
        String dni = sc.nextLine();
        boolean encontrado = false;
        for (String[] usuario : usuarios) {
            if (usuario[0].equals(dni)) {
                System.out.println("Usuario encontrado:");
                System.out.println(String.join(", ", usuario));
                encontrado = true;
                break;
            }
        }
        if (!encontrado)
            System.out.println("Usuario no encontrado.");
    }

    private static void mostrarMenu() {
        System.out.println("1. Agregar usuario");
        System.out.println("2. Eliminar usuario");
        System.out.println("3. Buscar usuario");
        System.out.println("4. Salir");
    }
    public static void main(String[] args) {
        cargarUsuarios();

        Scanner sc = new Scanner(System.in);
        int opcion;

        do {
            mostrarMenu();
            System.out.print("Ingrese el número de la opción que desea realizar: ");
            opcion = sc.nextInt();
            sc.nextLine(); // Limpiar el buffer

            switch (opcion) {
                case 1:
                    agregarUsuario(sc);
                    break;
                case 2:
                    eliminarUsuario(sc);
                    break;
                case 3:
                    buscarUsuario(sc);
                    break;
                case 4:
                    System.out.println("Saliendo del programa.");
                    break;
                default:
                    System.out.println("Opción no válida. Por favor, ingrese un número válido.");
            }
        } while (opcion != 4);
    }
}

File: PRO/practica45/practica45.py
usuarios = []

def cargar_usuarios():
    try:
        with open("usuarios.txt", "r") as archivo:
            for linea in archivo:
                userData = linea.strip().split(", ")
                usuarios.append(userData)
    except IOError:
        print("Error al cargar los usuarios.")

def guardar_usuarios():
    try:
        with open("usuarios.txt", "w") as archivo:
            for usuario in usuarios:
                archivo.write(", ".join(usuario) + "\n")
    except IOError:
        print("Error al guardar los usuarios.")

def agregar_usuario():
    dni = input("Ingrese el DNI del nuevo usuario: ")
    nombre = input("Ingrese el nombre del nuevo usuario: ")
    apellidos = input("Ingrese los apellidos del nuevo usuario: ")
    correo = input("Ingrese el correo electrónico del nuevo usuario: ")
    nuevo_usuario = [dni, nombre, apellidos, correo]
    usuarios.append(nuevo_usuario)
    guardar_usuarios()
    print("Usuario agregado correctamente.")

def eliminar_usuario():
    dni = input("Ingrese el DNI del usuario que desea eliminar: ")
    encontrado = False
    for usuario in usuarios:
        if usuario[0] == dni:
            usuarios.remove(usuario)
            guardar_usuarios()
            print("Usuario eliminado correctamente.")
            encontrado = True
            break
    if not encontrado:
        print("Usuario no encontrado.")

def buscar_usuario():
    dni = input("Ingrese el DNI del usuario que desea buscar: ")
    encontrado = False
    for usuario in usuarios:
        if usuario[0] == dni:
            print("Usuario encontrado:")
            print(", ".join(usuario))
            encontrado = True
            break
    if not encontrado:
        print("Usuario no encontrado.")

def mostrar_menu():
    print("1. Agregar usuario")
    print("2. Eliminar usuario")
    print("3. Buscar usuario")
    print("4. Salir")

def main():
    cargar_usuarios()

    while True:
        mostrar_menu()
        opcion = input("Ingrese el número de la opción que desea realizar: ")

        if opcion == "1":
            agregar_usuario()
        elif opcion == "2":
            eliminar_usuario()
        elif opcion == "3":
            buscar_usuario()
        elif opcion == "4":
            print("Saliendo del programa.")
            break
        else:
            print("Opción no válida. Por favor, ingrese un número válido.")

if __name__ == "__main__":
    main()
